crosshair/condition_parser.py

```diff
@@ -806,10 +814,103 @@ def get_class_invariants(self, cls: type) -> List[ConditionExpr]:
         return []
 
 
+class HypothesisParser(ConcreteConditionParser):
+
+    def get_fn_conditions(self, ctxfn: FunctionInfo) -> Optional[Conditions]:
+        fn_and_sig = ctxfn.get_callable()
+        if fn_and_sig is None:
+            return None
+        (fn, _) = fn_and_sig
+
+        # TODO replace this guard with package-level configuration?
+        if (
+                getattr(fn, "__module__", False)
+                and fn.__module__.startswith("crosshair.")
+                and not fn.__module__.endswith("_test")
+        ):
+            return None
+
+        try:
+            inner_test = fn.hypothesis.inner_test
+            inner_test_sig = inspect.signature(inner_test)
+        except AttributeError:
+            return None
+
+        given_kwargs = fn.hypothesis._given_kwargs
+        filename, first_line, _lines = sourcelines(fn)
+
+        pre: List[ConditionExpr] = []
+        post = [ConditionExpr(lambda _: True, filename, first_line, "")]
+
+        for variable, strategy in given_kwargs.items():
+            expr = self.get_expr_from_strategy(variable, strategy)
+            #     lower_bound = strategy.wrapped_strategy.start
+            #     upper_bound = strategy.wrapped_strategy.end
+            #     expr = f'isinstance({variable}, int)'
+            #
+            #     if lower_bound is not None and upper_bound is not None:
+            #         expr = f'{lower_bound} <= {variable} <= {upper_bound}'
+            #
+            #     elif lower_bound is not None:
+            #         expr = f'{lower_bound} <= {variable}'
+            #
+            #     elif upper_bound is not None:
+            #         expr = f'{variable} <= {upper_bound}'
+            if expr is not None:
+                condition_expr = condition_from_source_text(filename=filename,
+                                                            line=first_line,
+                                                            expr_source=expr,
+                                                            namespace=fn_globals(inner_test))
+                pre.append(condition_expr)
+
+        return Conditions(
+            fn=inner_test,
+            src_fn=inner_test,
+            pre=pre,  # (pre)
+            post=post,
+            raises=frozenset([hypothesis.errors.UnsatisfiedAssumption]),
+            sig=inner_test_sig,
+            mutable_args=None,
+            fn_syntax_messages=[],
+        )
+
+    def get_expr_from_strategy(self, variable, strategy: SearchStrategy) -> Optional[str]:
+        if isinstance(strategy, hypothesis.strategies._internal.strategies.OneOfStrategy):
+            strategy_list = strategy.original_strategies
+            expr = f'{self.get_expr_from_strategy(variable, strategy_list[0])}'
+            for arg_strategy in strategy_list[1:]:
+                expr += f' or {self.get_expr_from_strategy(variable, arg_strategy)}'
+            return expr
+
+        if str(strategy)[:8] == "integers":
```

Comment:
I suspect that this is not the correct way to identify an "integers" strategy, but for now just so i have working code ive left this in.

Reply:
`isinstance(strategy, hypothesis.strategies._internal.numbers.IntegersStrategy)`

Reply:
Im not sure why but `isinstance(strategy, hypothesis.strategies._internal.numbers.IntegersStrategy)` somehow doesn't recognise the integers strategy.

Reply:
That'll be because it's a LazyStrategy; use the unwrap function from hypothesis.strategies._internal.lazy to reify it before calling is instance.